### DISCUSSION TITLE
To get rid of this warning, please upgrade to bootstrap version >= 8

### DISCUSSION THREAD

**Discussion by anentropic** [aws/aws-cdk#27477] [Q&A]
When I run `cdk diff` I see a message like:

```
current credentials could not be used to assume 'arn:aws:iam::*********:role/cdk-hnb659fds-lookup-role-*********-eu-west-1', but are for the right account. Proceeding anyway.
(To get rid of this warning, please upgrade to bootstrap version >= 8)
current credentials could not be used to assume 'arn:aws:iam::*********:role/cdk-hnb659fds-deploy-role-*********-eu-west-1', but are for the right account. Proceeding anyway.
```

A couple of questions:
- are the credentials warnings and the bootstrap version warning related to each other, or separate warnings?
  - ...and how can it proceed anyway if it couldn't assume the role?
- how do I "upgrade to bootstrap version >= 8"?  what does it mean?

**Comment by anentropic** [comment]
> upgrade to bootstrap version >= 8

This seems to mean _"run `cdk bootstrap` to update the bootstrap stack in your aws account"_.

Weirdly I also have this message in my diff:

```
Other Changes
[+] Unknown Rules: {"CheckBootstrapVersion":{"Assertions":[{"Assert":{"Fn::Not":[{"Fn::Contains":[["1","2","3","4","5"],
{"Ref":"BootstrapVersion"}]}]},"AssertDescription":"CDK bootstrap stack version 6 required. Please run 'cdk bootstrap'
with a recent version of the CDK CLI."}]}}
```

...but I just ran `cdk bootstrap` and the "version >= 8" message went away, so presumably I have something newer than that.

**Comment by setjmp** [comment]
I had this problem running under WSL (Windows subystem for Linux).  It was caused by the Linux clock falling out of sync with actual time when my laptop suspended.

The remedy was to run `sudo hwclock -s`

More info:  https://stackoverflow.com/questions/65086856/wsl2-clock-is-out-of-sync-with-windows

**Comment by dennisvang** [comment]
There's one mention in seemingly unrelated [aws docs][1]:

>Error message | Solution 
>---------------------|---------------
>Current credentials could not be used to assume 'arn:aws:iam::[ACCOUNT_ID]:role/cdk-hnb659fds-file-publishing-role-[ACCOUNT_ID]-us-west-2', but are for the right account. Proceeding anyway. | The deployment account might not have been properly bootstrapped with CDK new style bootstrapping. Ensure the environment variable  `CDK_NEW_BOOTSTRAP` is set to `1` prior to bootstrapping your account. 

Also a [related question][2] on stackoverflow.

If you *are* using the latest bootstrap version and are seeing *only* the *credentials* warnings, without the *upgrade* warning, then it may be a permission issue. See [CDK wiki][3] for the required permissions.

[1]: https://docs.aws.amazon.com/solutions/latest/verifiable-controls-evidence-store/deployment.html#troubleshooting-common-deployment-issues
[2]: https://stackoverflow.com/q/71861264
[3]: https://github.com/aws/aws-cdk/wiki/Security-And-Safety-Dev-Guide#bootstrapping
